app/models/mobitel_delivery_detail.rb

```diff
@@ -0,0 +1,29 @@
+class MobitelDeliveryDetail < DeliveryDetail
+  # Mobitel APIs are not capable of providing delivery details.
+  # Hence we consider all the requests that passed the API validation as successful
+  def successful?
+    true
+  end
```

Comment:
[suggestion] Can we return `false` here? That is also equally misleading as a contract (the return status is neither `true` or `false`, it's `i don't know`.

But the benefit of returning false is that we seem to have code paths that are able to handle `successful`, `unsuccessful` and `in_progress` being all false and assuming an `unknown` communication result:
https://github.com/simpledotorg/simple-server/blob/1cee413645a3dc8baa90cfebc1e86e5214e015b7/app/models/communication.rb#L54-L64

But alas, this won't be enough—this kind of logic is not handled properly in the Experiment Notification model:
https://github.com/simpledotorg/simple-server/blob/f51d1510bafe2aa668e3b1e4ac31d7a282747d11/app/models/notification.rb#L68-L90

I wonder if we can fix the above logic. The result I'm trying to avoid is to have records in our DB that show a message is successfully sent when it might not be.

Reply:
I get your point. We can't say that a message is successful in binary anymore. However, with most SMS vendors, we rely on the information they provide as the truth.